File: python/demos/demo3_baseProgram/image.py
from sdk.common.Config import Config
from sdk.data.ProgramNode import ProgramNode
from sdk.Program import Program
from sdk.data.area.ImageNode import ImageNode
import json

def main():
    # with your own parameters
    # host = "${YourHost}"
    # sdk_key = "${YourSdkKey}"
    # sdk_secret = "${YourSdkSecret}"
    # Config.init_sdk(host, sdk_key, sdk_secret)

    # Build program data nodes
    image_path = "E:/Desktop/1b35.jpg"
    image_node = ImageNode(image_path)
    program_node = ProgramNode([image_node])

    # Instantiate program interfaces
    program = Program()

    # Add shows to devices
    device_ids = ["C16L-D00-A000F"]
    result = program.replace(device_ids, program_node)

    print("API Respond to the results:")
    print(json.dumps(result, indent=2))

if __name__ == "__main__":
    main()

File: java/demo/src/main/java/cn/huidu/device/demo/demo3_baseProgram/image.java
package cn.huidu.device.demo.demo3_baseProgram;

import cn.huidu.device.sdk.Program;
import cn.huidu.device.sdk.data.program.ProgramNode;
import cn.huidu.device.sdk.data.program.area.content.ImageNode;

public class image {
    public static void main(String[] args) {
        // Replace with your own parameters
        // String host = "${YourHost}";
        // String sdkKey = "${YourSdkKey}";
        // String sdkSecret = "${YourSdkSecret}";
        // Config.InitSdk(host, sdkKey, sdkSecret);

        // 1. Build program data node with only one image area
        ProgramNode programNode = new ProgramNode(
                new ImageNode("D:/WorkSpace/TestFiles/Images/1600829925_1836547993.dat.gif"));
        // 2. Instantiate program interface object
        Program program = new Program();

        // 3. Add program to device
        String jsonResult = program.replace(new String[] { "" }, programNode);
        // 4. Print result
        System.out.println(jsonResult);
    }

}
